### ISSUE TITLE
Azure / ARM steps failing to find resource groups

### ISSUE THREAD

**Issue by thebigmoosey** [OctopusDeploy/Issues#4844]
There has been an issue reported now by multiple customers who are trying to deploy to Azure, and the new calls to the Azure client libraries have been failing to return resource groups in the call to Azure Web Apps.

We either need to add retry logic to the command, and try again (to a maximum of X times) if we detect resource groups are missing in the response, and/or contact Azure to see if this is a known issue for customers in certain regions.

Source: https://help.octopus.com/t/could-not-find-azure-website-the-resource-group-name-is-empty/21149

**Comment by TomasMorton** [comment]
Hi @thebigmoosey,

This is affecting us majorly right now, as we have upgraded without realising that this issue existed.

The call to `WebSiteManagementClient.WebApps.List()` is frequently timing out with a TaskCanceledException.  When it does go through then a large (but random) portion of the sites do not have a resource group set.
For example, I ran this call twice and received:
1: 195 with a resource group, 1230 without a resource group.
2: 839 with a resource group, 586 without a resource group.

Can we possibly replace the call to `WebSiteManagementClient.WebApps.List()` with a call to `WebSiteManagementClient.WebApps.Get(string resourceGroupName, string name)`?  Locally, this call takes less than a second to complete so will greatly improve our deployment time.

This would be a change in behaviour though, because currently the resourceGroupName parameter in `ResourceManagerPublishProfileProvider.GetPublishProperties` is allowed to be null due to this check:
`(string.IsNullOrWhiteSpace(resourceGroupName) || string.Equals(webApp.ResourceGroup, resourceGroupName, StringComparison.InvariantCultureIgnoreCase)))`

Side note:  I also tried upgrading to version 2.0.0 of `Microsoft.Azure.Management.Websites`, but the issue still occurs there.
